watertap/unit_models/tests/test_uv_aop.py

```diff
@@ -207,32 +221,539 @@ def test_solve(self, UV_frame):
     @pytest.mark.component
     def test_solution(self, UV_frame):
         m = UV_frame
-        assert pytest.approx(0.999999999, rel=1e-3) == value(
+        assert pytest.approx(2052.999848078, rel=1e-3) == value(
             m.fs.unit.control_volume.properties_in[0].flow_mass_phase_comp["Liq", "H2O"]
         )
-        assert pytest.approx(0.001003, rel=1e-3) == value(
+        assert pytest.approx(2.0592, rel=1e-3) == value(
             m.fs.unit.control_volume.properties_in[0].flow_vol
         )
-        assert pytest.approx(74e-9, rel=1e-3) == value(
+        assert pytest.approx(1.51922e-4, rel=1e-3) == value(
             m.fs.unit.control_volume.properties_in[0].flow_mass_phase_comp[
                 "Liq", "NDMA"
             ]
         )
-        assert pytest.approx(0.999999999, rel=1e-3) == value(
+        assert pytest.approx(2053, rel=1e-3) == value(
             m.fs.unit.control_volume.properties_out[0].flow_mass_phase_comp[
                 "Liq", "H2O"
             ]
         )
-        assert pytest.approx(1.8248e-8, rel=1e-3) == value(
+        assert pytest.approx(4.8104e-5, rel=1e-3) == value(
             m.fs.unit.control_volume.properties_out[0].flow_mass_phase_comp[
                 "Liq", "NDMA"
             ]
         )
-        assert pytest.approx(5000, rel=1e-3) == value(m.fs.unit.uv_dose)
+        assert pytest.approx(500, rel=1e-3) == value(
+            pyunits.convert(m.fs.unit.uv_dose, to_units=pyunits.mJ / pyunits.cm**2)
+        )
         assert pytest.approx(0.04576, rel=1e-3) == value(m.fs.unit.UVA)
-        assert pytest.approx(1.3333e-4, rel=1e-3) == value(
-            m.fs.unit.reaction_rate_constant["Liq", "NDMA"]
+        assert pytest.approx(2.3e-3, rel=1e-3) == value(
+            m.fs.unit.photolysis_rate_constant["Liq", "NDMA"]
+        )
+        assert pytest.approx(1156.99, rel=1e-3) == value(
+            pyunits.convert(m.fs.unit.electricity_demand[0], to_units=pyunits.kW)
+        )
+
+    @pytest.mark.requires_idaes_solver
+    @pytest.mark.component
+    def test_costing(self, UV_frame):
+        m = UV_frame
+
+        m.fs.costing = WaterTAPCosting()
+        m.fs.costing.base_currency = pyunits.USD_2020
+
+        m.fs.unit.costing = UnitModelCostingBlock(
+            default={
+                "flowsheet_costing_block": m.fs.costing,
+            },
+        )
+        m.fs.costing.cost_process()
+        m.fs.costing.add_LCOW(m.fs.unit.control_volume.properties_out[0].flow_vol)
+        results = solver.solve(m)
+
+        assert results.solver.termination_condition == TerminationCondition.optimal
```

Comment:
You could replace this line with `assert_optimal_termination(results)`, which I think checks that `TerminationCondition.optimal` is true _and_ that solver status is OK